src/ProjectTemplates/ComponentsWebAssembly.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Client/Program.cs

```diff
@@ -19,7 +21,14 @@ public static async Task Main(string[] args)
             var builder = WebAssemblyHostBuilder.CreateDefault(args);
             builder.RootComponents.Add<App>("app");
 
+#if (!Hosted || NoAuth)
             builder.Services.AddSingleton(new HttpClient { BaseAddress = new Uri(builder.HostEnvironment.BaseAddress) });
+#else
+            builder.Services.AddHttpClient("ComponentsWebAssembly_CSharp.ServerAPI", client => client.BaseAddress = new Uri(builder.HostEnvironment.BaseAddress))
+                .AddHttpMessageHandler<BaseAddressAuthorizationMessageHandler>();
+            builder.Services.AddTransient(sp => sp.GetRequiredService<IHttpClientFactory>().CreateClient("ComponentsWebAssembly_CSharp.ServerAPI"));
```

Comment:
I understand this registers an `HttpClient`, but it's not obvious at first glance. People might think "hmm, I wonder what's registering the HttpClient type with DI" and not have a way to know it's this line, since the type name `HttpClient` doesn't appear anywhere in it.

To help people figure this out, how about prefixing it with a clarifying comment?

```suggestion
            // Supply HttpClient instances that include authorization tokens when making requests to the server project
            builder.Services.AddTransient(sp => sp.GetRequiredService<IHttpClientFactory>().CreateClient("ComponentsWebAssembly_CSharp.ServerAPI"));
```

Also, why is this registered as transient? On L25 (in the not-hosted-auth case) we register the `HttpClient` as singleton. Does something bad happen if we change L29 to use `AddSingleton`? If so, I'd love to know the details!

Reply:
I'm doing the same thing as httpclientfactory does. See
https://github.com/dotnet/runtime/blob/master/src/libraries/Microsoft.Extensions.Http/src/DependencyInjection/HttpClientFactoryServiceCollectionExtensions.cs#L57-L60

Also, I tried to use the default client that httpclientfactory registers, but for some reason it didn't work. I'm following up with ryan on that.

Reply:
done